### PR TITLE
fix(route): youtube playlistId

### PR DESCRIPTION
**Pull request by sohow**
## 该 PR 相关 Issue / Involved issue
Close #
https://github.com/DIYgod/RSSHub/issues/11576

## 完整路由地址 / Example for the proposed route(s)

```routes
/youtube/user/@tangshanlaowang
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [ ] 新的路由 New Route
  - [ ] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
fix youtube user rss bug

### CODE REVIEW

File: lib/v2/youtube/user.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 const utils = require('./utils');
 const config = require('@/config').value;
 const { parseDate } = require('@/utils/parse-date');
+// const {ca} = require("@/v2/embassy/supportedList"); //never used
+const got = require('@/utils/got');
```
Comment (TonyRL) [inline lib/v2/youtube/user.js:5]:
```suggestion
const got = require('@/utils/got');
```

File: lib/v2/youtube/user.js [hunk 2]
```diff
@@ -9,7 +12,21 @@ module.exports = async (ctx) => {
...
+        try {
+            const link = `https://www.youtube.com/@${username}`;
+            const response = await got(link);
+            const $ = cheerio.load(response.body);
+            const channelId = $('meta[itemprop="channelId"]').attr('content');
+            playlistId = (await utils.getChannelWithId(channelId, 'contentDetails', ctx.cache)).data.items[0].contentDetails.relatedPlaylists.uploads;
+        } catch (e) {
+            playlistId = ctx.params.username;
+        }
+    }
```
Comment (TonyRL) [inline lib/v2/youtube/user.js:29]:
```suggestion
    let playlistId;
    let channelName;
    if (username.startsWith('@')) {
        const link = `https://www.youtube.com/${username}`;
        const response = await got(link);
        const $ = cheerio.load(response.body);
        const channelId = $('meta[itemprop="channelId"]').attr('content');
        channelName = $('meta[itemprop="name"]').attr('content');
        playlistId = (await utils.getChannelWithId(channelId, 'contentDetails', ctx.cache)).data.items[0].contentDetails.relatedPlaylists.uploads;
    }
    playlistId = playlistId || (await utils.getChannelWithUsername(username, 'contentDetails', ctx.cache)).data.items[0].contentDetails.relatedPlaylists.uploads;
```

**Review by TonyRL — COMMENTED** [review]
Please update `ctx.state.data` as well
```js
        title: `${channelName || username} - YouTube`,
        link: channelName ? `https://www.youtube.com/${username}` : `https://www.youtube.com/user/${username}`,
```

**Review by sohow — COMMENTED** [review]
add "@username"

**Comment by sohow** [comment]
github-actions bot

**Review by sohow — COMMENTED** [review]
github-actions bot

**Pull request merged**